util/compare_test_results.py

```diff
@@ -0,0 +1,205 @@
+                if current_flat[test_path] == "PASS":
+                    fixes.append(test_path)
+
+    # Find newly skipped tests (were passing, now skipped)
+    newly_skipped = []
+    for test_path, status in current_flat.items():
+        if (
+            status == "SKIP"
+            and test_path in reference_flat
+            and reference_flat[test_path] == "PASS"
+        ):
+            newly_skipped.append(test_path)
+
+    # Find newly passing tests (were skipped, now passing)
+    newly_passing = []
+    for test_path, status in current_flat.items():
+        if (
+            status == "PASS"
+            and test_path in reference_flat
+            and reference_flat[test_path] == "SKIP"
+        ):
+            newly_passing.append(test_path)
+
+    return regressions, fixes, newly_skipped, newly_passing
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Compare GNU test results and identify regressions and fixes"
+    )
+    parser.add_argument("current_json", help="Path to current run JSON results")
+    parser.add_argument("reference_json", help="Path to reference JSON results")
+    parser.add_argument(
+        "--ignore-file",
+        required=True,
+        help="Path to file with tests to ignore (for intermittent issues)",
+    )
+    parser.add_argument("--output", help="Path to output file for GitHub comment")
+
+    args = parser.parse_args()
+
+    # Load test results
+    try:
+        with open(args.current_json, "r") as f:
+            current_results = json.load(f)
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        sys.stderr.write(f"Error loading current results: {e}\n")
+        return 1
+
+    try:
+        with open(args.reference_json, "r") as f:
+            reference_results = json.load(f)
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        sys.stderr.write(f"Error loading reference results: {e}\n")
+        sys.stderr.write("Skipping comparison as reference is not available.\n")
+        return 0
+
+    # Load ignore list (required)
+    if not os.path.exists(args.ignore_file):
+        sys.stderr.write(f"Error: Ignore file {args.ignore_file} does not exist\n")
+        return 1
+
+    ignore_list = load_ignore_list(args.ignore_file)
+    print(f"Loaded {len(ignore_list)} tests to ignore from {args.ignore_file}")
+
+    # Flatten result structures for easier comparison
+    current_flat = flatten_test_results(current_results)
+    reference_flat = flatten_test_results(reference_results)
+
+    # Identify different categories of test changes
+    regressions, fixes, newly_skipped, newly_passing = identify_test_changes(
+        current_flat, reference_flat
+    )
+
+    # Filter out intermittent issues from regressions
+    real_regressions = [r for r in regressions if r not in ignore_list]
+    intermittent_regressions = [r for r in regressions if r in ignore_list]
+
+    output_lines = []
+    have_new_failures = len(real_regressions) > 0
```

Comment:
This variable is unnecessary and you can use `real_regressions` instead (as you already do on line 157).